src/http/request.rs

```diff
@@ -49,7 +52,32 @@ impl<B> Request<B> {
     /// Read the Request body.
     #[inline]
     pub fn body(&self) -> Option<&B> { self.body.as_ref() }
-    
+
+    /// Take the Body of this Request
+    #[inline]
+    pub fn take_body(self) -> Option<B> { self.body }
```

Comment:
Yes, this looks like the best one:

```rust
pub fn body(self) -> Body { self.body.unwrap_or_default() }
```

Technically, with that it's possible for the impl to exist for `B: Default`, but I don't know if that helps anyone, really. For all incoming streams (incoming to this machine, built by hyper internall), it will always be `Body`.